engine/table/src/main/java/io/deephaven/engine/table/impl/by/FormulaMultiColumnChunkedOperator.java

```diff
@@ -0,0 +1,405 @@
+
+        if (!addsToProcess && !modifiesToProcess && !removesToProcess) {
+            // Nothing to do.
+            return;
+        }
+
+        if (!addsToProcess && !modifiesToProcess) {
+            // Only removes to handle, clear the removed objects.
+            if (!resultColumn.getType().isPrimitive()) {
+                try (final DataFillerContext dataFillerContext = new DataFillerContext()) {
+                    dataFillerContext.clearObjectColumnData(downstream.removed());
+                }
+            }
+            return;
+        }
+
+        try (final DataCopyContext dataCopyContext = new DataCopyContext()) {
+            // Clear removed objects from the result column.
+            if (removesToProcess && !resultColumn.getType().isPrimitive()) {
+                dataCopyContext.clearObjectColumnData(downstream.removed());
+            }
+            if (modifiesToProcess) {
+                dataCopyContext.copyData(downstream.modified());
+            }
+            if (addsToProcess) {
+                dataCopyContext.copyData(downstream.added());
+            }
+        }
+    }
+
+    @Override
+    public void propagateFailure(@NotNull final Throwable originalException,
+            @NotNull final TableListener.Entry sourceEntry) {
+        if (delegateToBy) {
+            groupBy.propagateFailure(originalException, sourceEntry);
+        }
+    }
+
+    @Override
+    public BucketedContext makeBucketedContext(final int size) {
+        return delegateToBy ? groupBy.makeBucketedContext(size) : null;
+    }
+
+    @Override
+    public SingletonContext makeSingletonContext(final int size) {
+        return delegateToBy ? groupBy.makeSingletonContext(size) : null;
+    }
+
+    /**
+     * Helper class to efficiently clear object data from the result columns.
+     */
+    private class DataFillerContext implements SafeCloseable {
+
+        final FillFromContext fillFromContext;
+
+        private DataFillerContext() {
+            fillFromContext = resultColumn.makeFillFromContext(BLOCK_SIZE);
+        }
+
+        void clearObjectColumnData(@NotNull final RowSequence rowSequence) {
+            try (final RowSequence.Iterator rowSequenceIterator = rowSequence.getRowSequenceIterator();
+                    final WritableObjectChunk<?, Values> nullValueChunk =
+                            WritableObjectChunk.makeWritableChunk(BLOCK_SIZE)) {
+                nullValueChunk.fillWithNullValue(0, BLOCK_SIZE);
+                while (rowSequenceIterator.hasMore()) {
+                    final RowSequence rowSequenceSlice = rowSequenceIterator.getNextRowSequenceThrough(
+                            calculateContainingBlockLastKey(rowSequenceIterator.peekNextKey()));
+                    nullValueChunk.setSize(rowSequenceSlice.intSize());
+                    resultColumn.fillFromChunk(fillFromContext, nullValueChunk, rowSequenceSlice);
+                }
```

Comment:
Be sure to propagate this to the older operator, too.
```suggestion
                    sharedContext.reset();
                }
```